daml/Marketplace/Distribution/Auction/Service.daml

```diff
@@ -85,7 +88,10 @@ template Service
 
           -- Generate Settlement instructions for allocated bids
           let settleAllocations = (\allocation -> Auction.SettleAllocation with price = finalPrice; issuer = customer; issuerReceivableAccount = receivableAccount; ..) <$> allocatedBids
-          (_, siCids) <- foldlA generateSettlementInstruction (Some depositCid, []) settleAllocations
+          (remainingDepositCidOpt, siCids) <- foldlA generateSettlementInstruction (Some depositCid, []) settleAllocations
+
+          -- Transfer remaining deposit back to the Issuer's trading account
+          whenSome remainingDepositCidOpt \depositCid -> (exerciseByKey @AllocationAccountRule allocationAccount.id Allocation.Withdraw with depositCid = depositCid; transferTo = tradingAccount) $> ()
```

Comment:
This is the bug I mentioned on slack from ex-bonds wasn't it? :grin:

Reply:
Not sure of the bug in `ex-bonds` (I only checked their dutch auction code) but the purpose of call is if the auction isn't fully filled (ie, there's a deposit existing from the issuer containing units) then return these units to their other normal account from their locked account.